### PR TITLE
doc: describe Unix signal handling better

### PR DESCRIPTION
**Pull request by vcunat**
- SIGTERM seems OK to handle - I tested that on Linux, read POSIX, etc.
  I can't see why it should be different from SIGUSR1, for example,
  except that the default handler is different.
- Also raise(15) is caught by the handler for me, and I can't see
  why it wouldn't.

The main source of knowledge for this is POSIX.
http://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_04_03_03

- - -
Or did some of the paragraphs I replaced mean to refer to Windows? (and not marked properly)  I can really only write about Unix systems.

### CODE REVIEW

**Comment by pspacek** [comment]
@piscisaureus Can this be merged after the last round of changes?

File: docs/src/signal.rst [hunk 1]
```diff
@@ -24,13 +27,26 @@ are never received. These signals are: `SIGILL`, `SIGABRT`, `SIGFPE`, `SIGSEGV`,
...
-
 .. versionchanged:: 1.15.0 SIGWINCH support on Windows was improved.
 
+Unix notes
+----------
+
+* SIGKILL and SIGSTOP are impossible to catch.
+
+* Handling SIGBUS, SIGFPE, SIGILL or SIGSEGV via libuv results into undefined behavior.
+  Also, calls to `abort()`, e.g. through `assert()`, will not be caught
```
[the remark this answers is not in the document]
Comment (sam-github) [reply docs/src/signal.rst]: Remove the vanilla C paragraph? Its unnecessary. The docs should describe uv, not generic C coding.

Or solve it some other way, but ATM, your sentence "Also, calls to `abort()`, .. and the process will terminate." is a non sequitor. I think you are talking about SIGABRT, but you don't say so. SIGABRT is different from the others, so why document it in the same bullet?

[the remark this answers is not in the document]
Comment (vcunat) [reply docs/src/signal.rst]: OK, I found it interesting how POSIX by itself forces this when deferring the handling code e.g. to an event loop, but I dropped the text now.  Feel free to squash the commits.

**Comment by vcunat** [comment]
So, something else to improve?

**Comment by vcunat** [comment]
I must say I feel a bit conflicted about doing (similar) contributions to libuv.

**Comment by bnoordhuis** [comment]
@vcunat Sorry, landed in 4fc5a0ee6 and thanks!

We frequently land pull requests in batches. Yours would eventually have been rounded up because it was already signed off on. :-)

**Comment by vcunat** [comment]
Ah, OK.  I didn't know the workflow, i.e. the fact that one positive review "is enough" and it only waits for such a roundup.

**Pull request closed without merging**